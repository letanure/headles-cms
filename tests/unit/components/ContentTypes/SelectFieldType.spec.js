import { shallowMount } from '@vue/test-utils'
import Component from '@/components/ContentTypes/SelectFieldType.vue'
require('babel-plugin-require-context-hook/register')()

describe(`${Component.name}.vue`, () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Component, {
      propsData: {
        value: 'text'
      },
      mocks: {
        $t: (msg) => msg,
      },
    })
  })

  describe('smoke tests', () => {
    it('should render', () => expect(wrapper.exists()).toBe(true))
    it('matches snapshot', () => expect(wrapper.element).toMatchSnapshot())
  })
})
