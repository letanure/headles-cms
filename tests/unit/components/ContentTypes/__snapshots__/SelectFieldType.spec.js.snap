// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFieldType.vue smoke tests matches snapshot 1`] = `
<el-select-stub
  autocomplete="off"
  class="SelectField"
  multiplelimit="0"
  placeholder="Select"
  popperappendtobody="true"
  value="text"
  valuekey="value"
>
  <el-option-stub
    label="fieldTyle.checkbox.label"
    value="checkbox"
  />
  <el-option-stub
    label="fieldTyle.color.label"
    value="color"
  />
  <el-option-stub
    label="fieldTyle.date.label"
    value="date"
  />
  <el-option-stub
    label="fieldTyle.email.label"
    value="email"
  />
  <el-option-stub
    label="fieldTyle.month.label"
    value="month"
  />
  <el-option-stub
    label="fieldTyle.number.label"
    value="number"
  />
  <el-option-stub
    label="fieldTyle.password.label"
    value="password"
  />
  <el-option-stub
    label="fieldTyle.radio.label"
    value="radio"
  />
  <el-option-stub
    label="fieldTyle.range.label"
    value="range"
  />
  <el-option-stub
    label="fieldTyle.tel.label"
    value="tel"
  />
  <el-option-stub
    label="fieldTyle.text.label"
    value="text"
  />
  <el-option-stub
    label="fieldTyle.time.label"
    value="time"
  />
  <el-option-stub
    label="fieldTyle.url.label"
    value="url"
  />
  <el-option-stub
    label="fieldTyle.week.label"
    value="week"
  />
</el-select-stub>
`;
